__tests__/app.test.js

```diff
@@ -257,4 +259,74 @@ describe("Challenge 6 - GET /api/articles/:article_id/comments", () => {
 
           
 
+})
+
+
+describe("Challenge 9 - DELETE /api/comments/:comment_id", () =>{
+
+    test("responds with a 204 and no content after deleting the specified article", () => {
+
+
+        return request(app)
+        .delete('/api/comments/4')
+        .expect(204)
+
+    })
+
+
+   
+    test("responds with a 204 and checks the requested comments is deleted", () => {
+
+        return request(app)
+        .delete('/api/comments/4')
+        .expect(204)
+        .then(() => {
+
+            return db.query(
+                `SELECT * FROM comments;`
+            ).then((body) => {
+                return body.rows
+            }).then((comments) => {
+                comments.forEach(comment => {
+
+                expect(comment).not.toHaveProperty('body', " I carry a log — yes. Is it funny to you? It is not to me." )
+
+                })
+            })
+
+        })
+
+
+    })
+
+    test('returns a 400 - when an invalid type is sent', () => {
+
+        return request(app)
+        .delete('/api/comments/NotAnId')
+        .expect(400)
+        .then(({body}) => {
+
+            expect(body).toMatchObject({
+                msg : "400 - invalid type request"
+            })
+        })
+
+    })
+
+
+    test('returns a 404 - when an incorrect article_id is given', () => {
+
+        return request(app)
+        .delete('/api/comments/9999')
+        .expect(404)
+        .then(({body}) => {
+
+
+            expect(body).toMatchObject({
+                msg : '404 - not found'
+            })
```

Comment:
toMatchObject is maybe not the best matcher here - to match object has the ability to ignore any properties you don't declare, it only cares that the ones you do declare do match. toEqual would check that the whole object matches as you declare. You could also do expect(body.msg).toBe('404 - not found')

You don't have to change this, just adding as a suggestion 😄